tradingstrategy/utils/groupeduniverse.py

```diff
@@ -728,15 +730,26 @@ def resample_candles(df: pd.DataFrame, new_timedelta: pd.Timedelta) -> pd.DataFr
         single_pair_candles = raw_candles.loc[raw_candles["pair_id"] == pair.pair_id]
         single_pair_candles = single_pair_candles.set_index("timestamp", drop=False)
         monthly_candles = resample_candles(single_pair_candles, TimeBucket.d30)
+        monthly_candles = resample_candles(single_pair_candles, TimeBucket.d30)
         assert len(monthly_candles) <= len(single_pair_candles) / 4
 
     """
     
     assert isinstance(new_timedelta, pd.Timedelta), f"We got {new_timedelta}, supposed to be pd.Timedelta. E.g. pd.Timedelta(hours=2)"
-    
+
+    # make sure float
+    if isinstance(df, pd.Series):
+        df = df.astype(float)
+
     #pandas_time_delta = new_bucket.to_pandas_timedelta()
     # https://stackoverflow.com/questions/21140630/resampling-trade-data-into-ohlcv-with-pandas
-    candles = df.resample(new_timedelta).mean(numeric_only=True)
+    candles = df.resample(new_timedelta).mean()
```

Comment:
I am not sure if this a correct way to resample, and it might be that open/high/etc. gets incorrectly resampled. I am not sure if this code path was ever correctly tested.

We are using a different method here:

https://github.com/tradingstrategy-ai/trade-executor/blob/6068ff60f9c0572ed1dee1b56e6d8c7e31e40f87/tradeexecutor/strategy/pandas_trader/alternative_market_data.py#L27

Can you confirm the resamping works by adding a simple test case that checks whether the given data was correctly resampled?